### PR TITLE
Support for v1 GenerateMask op ( dynamic shape ) 

### PR DESCRIPTION
**Pull request by nbpatel**

### CODE REVIEW

**Comment by nbpatel** [comment]
@diyessi @jbobba Do we need type_prop tests for experimental ops?

File: src/ngraph/op/experimental/generate_mask.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@
 using namespace std;
 using namespace ngraph;
 
-constexpr NodeTypeInfo op::GenerateMask::type_info;
+constexpr NodeTypeInfo op::v0::GenerateMask::type_info;
 
 #if 0
```
Comment (diyessi) [inline src/ngraph/op/experimental/generate_mask.cpp]: Kill this commented section.

File: src/ngraph/op/experimental/generate_mask.hpp [hunk 2]
```diff
@@ -24,17 +24,19 @@ namespace ngraph
...
+            {
+            public:
+                NGRAPH_API
+                static constexpr NodeTypeInfo type_info{"GenerateMask", 0};
+                const NodeTypeInfo& get_type_info() const override { return type_info; }
+                /// \brief Constructs a GenerateMask node with a given shape, seed,
+                /// probability and training/inference mode
+                GenerateMask() = default;
 
 #if 0
```
Comment (diyessi) [inline src/ngraph/op/experimental/generate_mask.hpp]: Kill this commented section.

**Comment by jbobba** [comment]
> @diyessi @jbobba Do we need type_prop tests for experimental ops?

Yes, we do.

File: src/ngraph/serializer.cpp [hunk 3]
```diff
@@ -1354,14 +1354,24 @@ shared_ptr<Node> JSONDeserializer::deserialize_node(json node_js)
...
-            node = make_shared<op::GenerateMask>(
-                args[0], output_shape, type, seed, probability, use_seed);
+            if (op_version == 0)
+            {
+                auto output_shape = node_js.at("output_shape").get<vector<size_t>>();
+
+                node = make_shared<op::v0::GenerateMask>(
+                    args[0], output_shape, type, seed, probability, use_seed);
+            }
+            if (op_version == 1)
```
Comment (gaurides) [inline src/ngraph/serializer.cpp]: should this be `if , else if`?

Comment (nbpatel) [reply src/ngraph/serializer.cpp]: shouldn;t matter, versioning is exclusive

**Pull request merged**